domain/src/main/java/org/oppia/android/domain/exploration/ExplorationProgressController.kt

```diff
@@ -154,6 +228,11 @@ class ExplorationProgressController @Inject constructor(
             )
           }
         } finally {
+          // If the answer was submitted on behalf of the Continue interaction, don't save
+          // checkpoint because it will be saved when the learner moves to the next state.
+          if (answerOutcome.state.interaction.id != "Continue")
```

Comment:
This seems hacky. There are probably more elegant alternatives, but before I get into that: do we actually need to special case this? Why not just treat it like any other interaction?

Reply:
If we do this like any other interaction, then checkpoints will be saved twice, once when the answer is submitted on behalf of continued interaction. and then the second time when the learner moves to the next state.

Reply:
Isn't that also true for other interactions? I'm not sure that I see an important distinction here for why 'Continue' should be handled differently.

Reply:
for any other interaction, after submitting the correct answer the user has  to press the continue button to go to the next state, so the user interacts with the UI twice before reaching the next state but with states with continue interaction, the user only interacts once with the UI when they press the continue button.

so since their are no user interactions between the two checkpoints of continue interaction, we can skip saving one checkpoint

Reply:
My concern is that this special case is a bit too specific. I think what might be helpful is if you introduce a helper method to check whether the interaction auto-continues and, if it doesn't, then you save the exploration checkpoint. That helper could be as simple as:

```kotlin
private fun doesInteractionAutoContinue(interactionId: String): Boolean =
  interactionId == "Continue"
```

This provides context as to why we're skipping it, and isolates the non-generic bit to a separate helper that could, in the future, be refactored if we add more auto-continuation interactions.

Reply:
added the helper function.